entity-framework/core/modeling/data-seeding.md

```diff
@@ -7,19 +7,74 @@ uid: core/modeling/data-seeding
 > This feature is new in EF Core 2.1.
 
-Data seeding allows to provide initial data to populate a database. Unlike in EF6, in EF Core, seeding data is associated with an entity type as part of the model configuration. Then EF Core [migrations](xref:core/managing-schemas/migrations/index) can automatically compute what insert, update or delete operations need to be applied when upgrading the database to a new version of the model.
+Unlike in EF6, in EF Core, seeding data can be associated with an entity type as part of the model configuration. Then EF Core [migrations](xref:core/managing-schemas/migrations/index) can automatically compute what insert, update or delete operations need to be applied when upgrading the database to a new version of the model.
 
-As an example, you can use this to configure seed data for a `Blog` in `OnModelCreating`:
+> [!NOTE]
+> Migrations only considers model changes when determining what operation should be performed to get the seed data into the desired state. Thus any changes to the data performed outside of migrations might be lost or cause an error.
+
+As an example, this will configure seed data for a `Blog` in `OnModelCreating`:
 
 [!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=BlogSeed)]
 
-To add entities that have a relationship the foreign key values need to be specified. Frequently the foreign key properties are in shadow state, so to be able to set the values an anonymous class should be used:
+To add entities that have a relationship the foreign key values need to be specified:
 
 [!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=PostSeed)]
 
-Once entities have been added, it is recommended to use [migrations](xref:core/managing-schemas/migrations/index) to apply changes. 
+If the entity type has any properties in shadow state an anonymous class can be used to provide the values:
+
+[!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=AnonymousPostSeed)]
+
+Owned entity types can be seeded in a similar fashion:
+
+[!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=OwnedTypeSeed)]
+
+See the [full sample project](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/DataSeeding) for more context.
+
+Once the data has been added to the model, [migrations](xref:core/managing-schemas/migrations/index) should be used to apply the changes.
+
+> [!TIP]
+> If you need to apply migrations as part of an automated deployment you can [create a SQL script](xref:core/managing-schemas/migrations/index#generate-sql-scripts) that can be previewed before execution.
 
 Alternatively, you can use `context.Database.EnsureCreated()` to create a new database containing the seed data, for example for a test database or when using the in-memory provider. Note that if the database already exists, `EnsureCreated()` will neither update the schema nor the seed data in the database.
+
```

Comment:
I think we should mention here that:
- If your database provider doesn't support migrations, then EnsureCreated() is the way to go
- If your database provider supports migration, you should not use EnsureCreated() if you want to use Migrations.